File: tests/integration/aliases/python/retype_component_child/step2/__main__.py
# Copyright 2016-2023, Pulumi Corporation.  All rights reserved.

import khulnasoft


class FooResource(khulnasoft.ComponentResource):
    def __init__(self, name, opts=None):
        alias_opts = khulnasoft.ResourceOptions(aliases=[khulnasoft.Alias(type_="my:module:FooResource")])
        opts = khulnasoft.ResourceOptions.merge(opts, alias_opts)
        super().__init__("my:module:FooResourceNew", name, None, opts)


class ComponentResource(khulnasoft.ComponentResource):
    def __init__(self, name, opts=None):
        super().__init__("my:module:ComponentResource", name, None, opts)
        FooResource("child", khulnasoft.ResourceOptions(parent=self))


ComponentResource("comp")

File: tests/integration/aliases/go/retype_component_child/step2/main.go
// Copyright 2016-2023, Pulumi Corporation.  All rights reserved.
//go:build !all
// +build !all

package main

import (
	"github.com/khulnasoft/khulnasoft/sdk/v3/go/khulnasoft"
)

type FooResource struct {
	khulnasoft.ResourceState
}

type FooComponent struct {
	khulnasoft.ResourceState
}

func NewFooResource(ctx *khulnasoft.Context, name string, opts ...khulnasoft.ResourceOption) (*FooResource, error) {
	fooRes := &FooResource{}
	aliasOpt := khulnasoft.Aliases([]khulnasoft.Alias{{
		Type: khulnasoft.String("my:module:FooResource"),
	}})
	opts = append(opts, aliasOpt)
	err := ctx.RegisterComponentResource("my:module:FooResourceNew", name, fooRes, opts...)
	if err != nil {
		return nil, err
	}
	return fooRes, nil
}

func NewFooComponent(ctx *khulnasoft.Context, name string, opts ...khulnasoft.ResourceOption) (*FooComponent, error) {
	fooComp := &FooComponent{}
	err := ctx.RegisterComponentResource("my:module:FooComponent", name, fooComp, opts...)
	if err != nil {
		return nil, err
	}
	parentOpt := khulnasoft.Parent(fooComp)
	_, err = NewFooResource(ctx, "child", parentOpt)
	if err != nil {
		return nil, err
	}
	return fooComp, nil
}

func main() {
	khulnasoft.Run(func(ctx *khulnasoft.Context) error {
		_, err := NewFooComponent(ctx, "foo")
		if err != nil {
			return err
		}

		return nil
	})
}
